extensions/amp-auto-ads/0.1/amp-auto-ads.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev, user} from '../../../src/log';
+import {xhrFor} from '../../../src/xhr';
+import {getConfigUrl} from './config-url';
+
+/** @const */
+const TAG = 'amp-auto-ads';
+
+export class AmpAutoAds extends AMP.BaseElement {
+
+  /** @override */
+  createdCallback() {
+    const type = this.element.getAttribute('type');
+    if (!type) {
+      dev().error(TAG, 'Required attribute type.');
```

Comment:
this is a user config error, pls use `user()`

also, if you want to fail quickly, just do `user().assert(type, 'Missing type attribute');`